extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/security/HttpAuthenticationMechanism.java

```diff
@@ -21,10 +22,16 @@ public interface HttpAuthenticationMechanism {
     Uni<ChallengeData> getChallenge(RoutingContext context);
 
     /**
-     * Returns the required credential types. If there are no identity managers installed that support the
-     * listed types then this mechanism will not be enabled.
+     * If this mechanism delegates authentication to the {@link IdentityProviderManager}, then the mechanism should
+     * signal with which {@link IdentityProvider#getRequestType()} is
+     * {@link IdentityProviderManager#authenticate(AuthenticationRequest)}
+     * called. This way, Quarkus can validate that required {@link IdentityProvider} exists and fail if it doesn't.
```

Comment:
```suggestion
     * It allows Quarkus to validate that one or more {@link IdentityProvider} providers with matching supported {@link IdentityProvider#getRequestType()} request types exist and fail otherwise.
```